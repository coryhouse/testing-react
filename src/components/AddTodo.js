import React from 'react';
import { Form, Button } from 'react-bootstrap';

const AddTodo = props => {
  return (
    <Form onSubmit={props.handleAddTodo}>
      <Form.Group controlId="item">
        <Form.Label>What you want to do?</Form.Label>
        <Form.Control type="text" name="todo" placeholder="Enter todo" />
      </Form.Group>
      <Button variant="primary" type="submit">
        Add Todo
      </Button>
    </Form>
  );
};

export default AddTodo;
