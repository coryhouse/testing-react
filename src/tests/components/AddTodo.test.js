import React from 'react';
import { shallow } from 'enzyme';
import AddTodo from '../../components/AddTodo';


test('should handle onSubmit handler', () => {
    const onSubmitSpy = jest.fn();
    const wrapper = shallow(<AddTodo handleAddTodo={onSubmitSpy} />);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('Form').simulate('submit');
    expect(onSubmitSpy).toHaveBeenCalled();
});
