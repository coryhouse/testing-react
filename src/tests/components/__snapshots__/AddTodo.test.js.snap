// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle onSubmit handler 1`] = `
<Form
  inline={false}
  onSubmit={[MockFunction]}
>
  <FormGroup
    controlId="item"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      What you want to do?
    </FormLabel>
    <FormControl
      name="todo"
      placeholder="Enter todo"
      type="text"
    />
  </FormGroup>
  <Button
    active={false}
    disabled={false}
    type="submit"
    variant="primary"
  >
    Add Todo
  </Button>
</Form>
`;
